sui-execution/latest/sui-adapter/src/temporary_store.rs

```diff
@@ -483,6 +483,26 @@ impl<'backing> TemporaryStore<'backing> {
         system_state_wrapper.storage_rebate = unmetered_storage_rebate;
         self.write_object(system_state_wrapper, WriteKind::Mutate);
     }
+
+    /// Given a modified object ID, return its metadata.
+    /// A modified object must be either a mutable input, or a loaded child object.
+    /// The only exception is when we upgrade system packages, in which case the upgraded
+    /// system packages are not part of input, but are modified.
+    fn get_object_modified_at(&self, object_id: &ObjectID) -> (VersionDigest, Owner) {
+        self.mutable_input_refs
+            .get(object_id)
+            .cloned()
+            .or_else(|| {
+                self.loaded_child_objects
+                    .get(object_id)
+                    .map(|metadata| ((metadata.version, metadata.digest), metadata.owner))
+            })
+            .unwrap_or_else(|| {
+                assert!(is_system_package(*object_id));
```

Comment:
A tiny bit nervous about the naked assert here, but should be fine

Reply:
Actually, is this a bigger issue in the presence of dev inspect?

Reply:
Shouldn't be? Even for dev_inspect, any modified object is either in the mutable inputs, or a loaded child object.

Reply:
What if I pass in the bytes for an object directly?

Reply:
I think it will be marked as unwrapped in that case, but just checking

Reply:
> What if I pass in the bytes for an object directly?

Could you elaborate on what this means?